### PR TITLE
(port) Removes an unneeded piece of virology RNG

### PR DESCRIPTION
**Pull request by boodaliboo**
original pr: https://github.com/tgstation/tgstation/pull/53879
"Okay, so, this is gonna take a bit of explanation.

Under our current virology system, when you add a symptom to a virus that doesn't already have that symptom, the game chooses a number: either 4, 5, or 6. If the number of symptoms in that virus BEFORE adding that new symptom is NOT less than (apparently, >= was a foreign concept to the person who wrote this code) the number chosen, the virus loses a random symptom before the new one is added.

This changes that check to work much more sanely and with less RNG. Basically, the number of symptoms in your virus is capped at a flat 6 now, and a random symptom will be removed from your virus only if adding a new symptom would take you over that cap.

To put it more simply, this changes a if(!(symptoms.len < (VIRUS_SYMPTOM_LIMIT - 1) + rand(-1, 1))) check to be a if(symptoms.len >= VIRUS_SYMPTOM_LIMIT) check."
#### Changelog

:cl:  
rscdel: Removed stupid viro rng
/:cl:

### CODE REVIEW

**Comment by Bibby0110** [comment]
your code is bad

**Comment by JamieD1** [comment]
@boodaliboo The code has differences between tg and ours. Double check what they do, this is not a simple copy paste im afraid.

**Comment by boodaliboo** [comment]
time to test a pr for the first time

File: code/datums/diseases/advance/advance.dm [hunk 1]
```diff
@@ -360,7 +360,7 @@
 	if(HasSymptom(S))
 		return
 
-	if(symptoms.len < (VIRUS_SYMPTOM_LIMIT - 1) + rand(-1, 1))
+	if(symptoms.len >= VIRUS_SYMPTOM_LIMIT)
 		symptoms += S
 	else
 		RemoveSymptom(pick(symptoms))
```
Comment (JamieD1) [inline code/datums/diseases/advance/advance.dm]:
```suggestion
	if(symptoms.len >= VIRUS_SYMPTOM_LIMIT)
		RemoveSymptom(pick(symptoms))
	symptoms += S
```

**Review by JamieD1 — APPROVED** [review]
